airframe-sql/src/main/scala/wvlet/airframe/sql/parser/SQLInterpreter.scala

```diff
@@ -399,12 +399,16 @@ class SQLInterpreter(withNodeLocation: Boolean = true) extends SqlBaseBaseVisito
     val alias = Option(ctx.AS())
       .map(x => expression(ctx.identifier()))
       .orElse(Option(ctx.identifier()).map(expression(_)))
-    SingleColumn(expression(ctx.expression()), None, getLocation(ctx))
+    val child = expression(ctx.expression())
+    val qualifier = child match {
+      case a: Attribute => a.qualifier
+      case _            => None
+    }
+    SingleColumn(child, qualifier, getLocation(ctx))
```

Comment:
The true problem was that the qualifier was not properly set when generating LogicalPlan. This fix ensures that presence or absence of qualifiers in select ... columns